File: APCS202306_路徑偵測.py
# 0:e ; 1:s ; 2:w ; e:n
n = int(input())
predir = 0
le = ri = rev = 0
prex, prey = map(int, input().split())
for i in range(1, n):
    x, y = map(int, input().split())
    if x > prex:
        dir = 0
    elif y < prey:
        dir = 1
    elif x < prex:
        dir = 2
    else:
        dir = 3

    checkdir = (dir-predir) % 4
    if checkdir == 1:
        ri += 1
    elif checkdir % 4 == 2:
        rev += 1
    elif checkdir % 4 == 3:
        le += 1

    predir, prex, prey = dir, x, y

print(le, ri, rev)

File: APCS202306_路徑偵測.cpp
#include <bits/stdc++.h>
using namespace std;

#define N 100010
typedef long long ll;

int main()
{
    // freopen("Q_4_10_3.in", "r", stdin);
    int n, prex, prey, x, y, predir, dir;
    int le = 0, ri = 0, rev = 0;
    cin >> n;
    cin >> prex >> prey;
    predir = 0; // 0:e ; 1:s ; 2:w ; 3:n
    for (int i = 1; i < n; ++i)
    {
        cin >> x >> y;
        if (x > prex)
            dir = 0;
        else if (y < prey)
            dir = 1;
        else if (x < prex)
            dir = 2;
        else
            dir = 3;

        int checkdir = (dir - predir + 4) % 4;
        if (checkdir == 1)
            ri++;
        else if (checkdir == 2)
            rev++;
        else if (checkdir == 3)
            le++;

        predir = dir, prex = x, prey = y;
    }
    cout << le << ' ' << ri << ' ' << rev;
    return 0;
}
